lib/reply.js

```diff
@@ -170,7 +170,9 @@ function onSendEnd (reply, payload) {
     return reply.res
   }
 
-  if (!reply.res.getHeader('Content-Type') || reply.res.getHeader('Content-Type') === 'application/json') {
+  if (!reply.res.getHeader('Content-Type') && typeof payload === 'string') {
```

Comment:
There are several instances of `reply.res.getHeader('Content-Type')` in this function. I think it would be wise to cache the lookup, if for no other reason than for maintainability.